test/Graph/Test_graph.cpp

```diff
@@ -21,4 +23,19 @@ TEST_CASE("Graph") {
     // CHECK(graph.adjMatrix()->contains(1, 2));
     // CHECK(graph.adjMatrix()->contains(2, 1));
   }
-}
\ No newline at end of file
+
+  SUBCASE("Constructor_2") {
+    SparseMatrix sm(4, 4);
+    sm.insert(0, 1, true);
+    sm.insert(1, 0, true);
+    sm.insert(1, 2, true);
+    sm.insert(2, 3, true);
+    sm.insert(3, 2, true);
+    Graph graph{sm};
+    CHECK(graph.nodeSet().size() == 4);
+    CHECK(graph.streetSet().size() == 5);
+    CHECK(graph.adjMatrix()->size() == 5);
+    // CHECK(graph.adjMatrix()->contains(1, 2));
```

Comment:
We're building it from adjMatrix so we should be able to check it, uncommenting these two lines